common/task/task.go

```diff
@@ -0,0 +1,28 @@
+package task
+
+import "time"
+
+type StopFn func()
+
+func RunTaskRepeateadly(task func(), period time.Duration) StopFn {
+	// Setup the ticket and the channel to signal
+	// the ending of the interval
+	ticker := time.NewTicker(period)
+	stop := make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				task()
+			case <-stop:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	return func() {
+		close(stop)
```

Comment:
I think calling this function won't ever cause the `ticker.Stop()` block to be run, I played around with this here: https://play.golang.org/p/DmOIKBLPLFR

Reply:
you're right...
closing a channel is a way to stop things but not here!
Thanks!